src/data/fin.lean

```diff
@@ -355,7 +374,7 @@ def cast (eq : n = m) : fin n ≃o fin m :=
 
 @[simp] lemma symm_cast (h : n = m) : (cast h).symm = cast h.symm := rfl
 
-@[simp] lemma coe_cast (h : n = m) (i : fin n) : (cast h i : ℕ) = i := rfl
+lemma coe_cast (h : n = m) (i : fin n) : (cast h i : ℕ) = i := rfl
```

Comment:
What makes this undesirable as a simp lemma?

Reply:
The PR adds a more general simp lemma about any `order_iso` from `fin n` to `fin m`.